### ISSUE TITLE
Can't import exported shared library back to penpot

### ISSUE THREAD

**Issue by andrewpiskunov** [penpot/penpot#1197]
**Describe the bug**
If you donwload penpot default library and try to import it back to penpot or to self-hosted instanse your upload stucks without any warning.

**To Reproduce**
1.  Go to Shared Libraries
2.  Export "Penpot - Desing system" lib
3.  Try to import the lib back to penpot or to self-hosted instanse

**Expected behavior**
The upload successfuly finished. 

**Screenshots**
Infinity uploading.
![image](https://user-images.githubusercontent.com/45717261/132739712-a15e67ac-60ff-4291-9d83-e50eac00ed1e.png)

Dev-tools console error.
![image](https://user-images.githubusercontent.com/45717261/132739902-8f9cd981-bf78-4fff-9b28-ce87b74f12bd.png)

**Environment (please complete the following information):**
It happends both - in SAAS and self-hosted versions.

**Comment by Igorgro** [comment]
Having the same problem, but I found a workaround. If the file imported not through drag-and-drop to "Shared libraries", but imported in "Project", it successfully uploaded and then appears as shared library.

**Comment by Alotor** [comment]
Thanks for the report @piskun0f 

The problem is that some of the images in the library are "broken" (they reference URL's that are no longer present). I've done a fix that will be deployed today that will fix the issue. If you're using a self-hosted instance you will need to upgrade to the 1.8 release (we'll be pushing it today).

Sorry for the inconvenience.

**Comment by viviicat** [comment]
I seem to be still seeing this in the 1.8.2 docker image.

**Comment by codedChaos** [comment]
Seeing the same issue using the online hosted version https://design.penpot.app/

**FWIW** -
It happens when trying to drag and drop into the 'Shared Library' page. It never seems to start processing the files in any of the library exports, it just hangs indefinitely. 

**Good news** is that it seems to be working fine when I simply import it into the project itself. This appears to be in line with this users post: https://github.com/penpot/penpot/issues/1197#issuecomment-916424902

> [Having the same problem, but I found a workaround. If the file imported not through drag-and-drop to "Shared libraries", but imported in "Project", it successfully uploaded and then appears as shared library.](https://github.com/penpot/penpot/issues/1197#issuecomment-916424902)

**Comment by shawngmc** [comment]
If the issue is actually drag-and-drop vs. add button w/file picker: As a workaround/UI consistency improvement, would adding an "Add" button to pop up a file selector work?

Otherwise, is there any way to get a more useful error out of the Penpot Frontend (for example, if this is some sort of front-end state issue)?

**Comment by viviicat** [comment]
What's odd, is drag-n-drop seems to work from other pages. It's specifically the Shared Library window that trips it up. On my install at least.

**Comment by viviicat** [comment]
I notice that the `api/rpc/mutation/create-temp-file` api post is failing with a spec validation failure.

The request looks like this:
```
[
  "^ ",
  "~:id",
  "~ue6cddda0-8572-11ec-a3dc-cb76e18b153d",
  "~:name","Whiteboarding & mapping kit",
  "~:is-shared",
  true,
  "~:project-id",
  null,
  "~:data",
  [...]
]
```

Of course, the equivalent request when uploading to one of the project dirs which succeeds has a project ID set.

Could the request be failing because the project-id is null because the user is not currently in a project when importing from this page? If I'm reading the code correctly, [this line of that RPC](https://github.com/penpot/penpot/blob/c27d709b6bd3612c671b9ef3a80a8950ff474471/backend/src/app/rpc/mutations/files.clj#L46) is attempting to look for the permissions of the project, which presumably can't be null.

**Comment by viviicat** [comment]
Deleting my investigation comment as it appears this may be fixed in pull request #1363 and I guess the Docker image is just not updated to include this change yet.

**Comment by niwinz** [comment]
> If the issue is actually drag-and-drop vs. add button w/file picker: As a workaround/UI consistency improvement, would adding an "Add" button to pop up a file selector work?

You already have the button as replacement for drag and drop and it is there for a while.

![image](https://user-images.githubusercontent.com/843689/152493476-3af682db-a5ce-4f37-b71c-5f784cb462bb.png)

You don't have this button on shared-library section because shared libraries is a "search of all files of the team that are marked as shared library". **You can't import here nothing because you only can import files to projects, the shared libraries does not belongs to any project because is a filter of all shared libraries from all projects of the team**

If you want import penpot file, go to project and import it.
If you have a concrete problem with some file, please open an other issue or write tu support.
